Add capability to use multiple stubs

duonoid opened the pull request:
# Overview
Add ability to use multiple stubs.

## Details
* previous implementation only considered 1 stub, but there are times
  when multiple stubs are used within 1 test/spec
* GCR should be told about stubs *before* `start_playing` or
  `start_recording` (i.e., before `with_cassette`) so that requests can be intercepted and recorded
  properly
* ~~improve cleanup by undef'ing `#orig_request_response`, which also
  asserts that the method exists (was intercepted properly)~~
  
  ## Limitations
 * no way to easily remove/reset stubs
 * no additional specs added

duonoid committed:
Add capability to use multiple stubs

* previous implementation only considered 1 stub, but there are times
  when multiple stubs are used within 1 test/spec
* GCR should be told about stubs *before* start_playing or
  start_recording so that requests can be intercepted and recorded
  properly
* improve cleanup by undef'ing #orig_request_response, which also
  asserts that the method exists (was intercepted properly)
* observation: streaming requests are not recorded

carsol commented on lib/gcr/cassette.rb:
```diff
@@ -77,46 +77,56 @@ def play(&blk)
+    GCR.stubs.each do |klass|
+      klass.class_eval do
+        alias_method :request_response, :orig_request_response
+        undef :orig_request_response
```
Curious why do you need to `undef :orig_request_response`?

duonoid replied:
After thinking about it, the patch would be a little simpler without the `undef`, so I have removed.  Originally, I was thinking of it as a matter of cleanup/symmetry (since `cassette` defined the method, it also removed it). I did some local testing without the `undef` and code seemed to behave appropriately.

carsol replied:
👍

duonoid committed:
Simplify by removing undef :orig_request_response

* original thought was a matter of cleanup, but after consideration and
  PR comment, the patch should be simpler without it (i.e., in case an
  external user defined :orig_request_response, we don't want to remove
  it)
